java-type-checker/java_type_checker/types.py

```diff
@@ -133,25 +140,49 @@ def method_named(self, name):
                     pass
             raise NoSuchJavaMethod("{0} has no method named {1}".format(self.name, name))
 
+    def is_subtype_of(self, other):
+        if other == self:
+            return True
+        for supertype in self.direct_supertypes:
+            if supertype.is_subtype_of(other):
+                return True
+        return False
+
 
 class JavaVoidType(JavaType):
     """The Java type `void`.
 
     It is never legal to use the result of a method returning void inside a larger expression.
     Void is therefore subtype only of itself, and not any other type.
     """
+
     def __init__(self):
         super().__init__("void")
 
+    def is_subtype_of(self, other):
+        if other == self:
```

Comment:
I'm not sure why yours didn't include this, but anyway, this is another example of `return other == self` being better.

Reply:
Mine does include it. Maybe you were looking at the “without bonus solutions” link?

Reply:
Yep, I missed that that's a bonus thing.